python/tk_multi_publish2/dialog.py

```diff
@@ -668,9 +668,36 @@ def _create_item_details(self, tree_item):
             self.ui.context_widget.show()
 
             if item.context_change_allowed:
-                self.ui.context_widget.enable_editing(
-                    True, "<p>Task and Entity Link to apply to the selected item:</p>"
-                )
+                # Check for task_required and change UI styling accordingly
+                if not self._validate_task_required():
+                    # Change task label color to SG_ALERT_COLOR
+                    self.ui.context_widget.ui.task_label.setStyleSheet(
+                        "color: "
+                        + sgtk.platform.current_bundle().style_constants[
+                            "SG_ALERT_COLOR"
+                        ]
+                    )
+                    # Also change the text and color of the label
+                    # Use SG_HIGHLIGHT_COLOR for better readability
+                    self.ui.context_widget.enable_editing(
+                        True,
+                        "<p>Task Required is <b>True</b> in your configuration. "
+                        "Please select a Task to continue.</p>",
+                    )
+                    self.ui.context_widget.ui.label.setStyleSheet(
```

Comment:
I think you should be able to use `context_widget.context_label` instead of going via the `ui` attribute?
I'm uncomfortable with us using the `.ui` property as that is not intended to be public .

Reply:
Also could the color be set through the `enable_editing`, via html tags? That would save you having to modify the stylesheet.